src/DSPC.sol

```diff
@@ -0,0 +1,293 @@
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /// @notice Add a facilitator
+    /// @param usr The address to add as a facilitator
+    /// @dev Sets buds[usr] to 1 and emits Kiss event. Facilitators can propose rate updates but cannot modify system parameters
+    function kiss(address usr) external auth {
+        buds[usr] = 1;
+        emit Kiss(usr);
+    }
+
+    /// @notice Remove a facilitator
+    /// @param usr The address to remove as a facilitator
+    /// @dev Sets buds[usr] to 0 and emits Diss event
+    function diss(address usr) external auth {
+        buds[usr] = 0;
+        emit Diss(usr);
+    }
+
+    /// @notice Configure module parameters
+    /// @param what The parameter to configure
+    /// @param data The value to set
+    /// @dev Emits File event after successful configuration
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "bad") {
+            require(data <= 1, "DSPC/invalid-bad-value");
+            bad = data;
+        } else if (what == "tau") {
+            tau = data;
+        } else {
+            revert("DSPC/file-unrecognized-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /// @notice Configure constraints for a rate
+    /// @param id The rate identifier (ilk name, "DSR", or "SSR")
+    /// @param what The parameter to configure ("min", "max" or "step")
+    /// @param data The value to set
+    /// @dev Emits File event after successful configuration
+    function file(bytes32 id, bytes32 what, uint256 data) external auth {
+        require(data <= type(uint16).max, "DSPC/invalid-value");
+        if (what == "min") {
+            require(data <= _cfgs[id].max, "DSPC/min-too-high");
+            _cfgs[id].min = uint16(data);
+        } else if (what == "max") {
+            require(data >= _cfgs[id].min, "DSPC/max-too-low");
+            _cfgs[id].max = uint16(data);
+        } else if (what == "step") {
+            _cfgs[id].step = uint16(data);
+        } else {
+            revert("DSPC/file-unrecognized-param");
+        }
+
+        emit File(id, what, data);
+    }
+
+    /// @notice Apply rate updates
+    /// @param updates Array of rate updates to apply
+    /// @dev Each update is validated against configured constraints before being applied
+    /// @dev Emits Set event after all updates are successfully applied
+    /// @dev Reverts if:
+    ///      - Empty updates array
+    ///      - Cooldown not expired
+    ///      - Rate below configured minimum
+    ///      - Rate above configured maximum
+    ///      - Rate change above configured step
+    function set(ParamChange[] calldata updates) external toll good {
+        require(updates.length > 0, "DSPC/empty-batch");
+
+        for (uint256 i = 0; i < updates.length; i++) {
+            bytes32 id = updates[i].id;
+            uint256 bps = updates[i].bps;
+
+            if (block.timestamp >= tau + _cfgs[id].toc) _sync(id);
```

Comment:
Nit: I personally prefer to keep the absolute timestamp before the interval when adding them.
```suggestion
            if (block.timestamp >= _cfgs[id].toc + tau) _sync(id);
```